pages/api/cep/v1/[cep].js

```diff
@@ -1,5 +1,6 @@
 import microCors from 'micro-cors';
 import cep from 'cep-promise';
+import rateLimit from '../../../../util/rateLimit';
```

Comment:
Sugestão:

O que acha de criar o rate limit como decorador http? Assim ele poderia ser aplicado sobre o item do `export default`

Reply:
Então Danilo eu não entendi bem! Você poderia mostrar uma poc disso?
Como você disse "sobre o item do export default", eu entendi que seria como um "HOC", confirma?

Reply:
> Sugestão:
> 
> O que acha de criar o rate limit como decorador http? Assim ele poderia ser aplicado sobre o item do `export default`

Estou bem mais familizarizado com o express, consigo usar os middlewares de forma mais fácil! A ideia que tenho sobre o assunto "rate limit", é verificar as requisições idependente do endpoint chamado.
no caso que implementei, fiquei bem confuso em entender como é resolvida a requisição dentro no NextJs (express temos req,res e next xD) um cadinho mais fácil. Por esse motivo deixei ele simples dentro da página do CEP.

Reply:
Seria um HOC. De certa forma, há uma leve semelhança com o middleware, visto q há delegação para o item decorado após validação.

Se vc me permitir, posso adicionar um commit a mais com a diferença. Ou poderia criar um gist. O que for melhor para vc.

Reply:
Rate limit em Rest APIs faz mais sentido por recurso. Mas, não sei qual direção @lucianopf e @filipedeschamps pensam em seguir.

Reply:
Oi @kkkasio , acabei criando um PR sobre o seu branch para exemplificar. Se você achar válido, você pode fazer o merge e este PR será atualizado.

Seria interessante considerar adicionar testes relacionados ao rate-limit.

Reply:
@daniloster Bacana a implementação! Eu sinceramente gostei, foi a forma que tinha pensado inicialmente. Porém olhando o resultado final, sinceramente **eu** não gosto de como fica o componente final que "implementa" o HOC _HOC(component(props))_ .
Na minha visão o desing fica bem mais bonito, manipulando a requisição de algum modo antes de renderizar a págiina (middleware xD).

Acho bacana esperar a visão dos Autores, assim podemos alinhar e definir melhor essa feature 🚀

Reply:
Tranquilo. Gostaria de ressaltar que o rate limit não pertence ao domínio do http-handler/resolver. Ele é um requisito não funcional e, portanto, é importante desacoplar ele para escalabilidade do sistema. 😉